src/main.rs

```diff
@@ -560,11 +592,23 @@ impl Application for Halloy {
             Screen::Migration(migration) => migration.view().map(Message::Migration),
         };
 
-        container(content)
+        let modal: Option<Element<'_, _>> = match &self.modal {
+            Modal::None => None,
+            Modal::ReloadConfigurationError(error) => Some(
+                modal::reload_configuration_error::view(error).map(|message| Message::Modal(message.into()))
+            ),
+        };
+
+        let content = container(screen)
             .width(Length::Fill)
             .height(Length::Fill)
             .style(theme::container::primary)
-            .into()
+            .into();
+
+        match modal {
+            Some(modal) => widget::modal(content, modal).into(),
```

Comment:
I'll look into this, but we don't want to change the shape of view at this root level because all our text caching lives in the state tree.

When view changes and state tree is diffed, it'll cause everything to get dropped and recreated (expensive!!)

Ideally we have a way to toggle the modal dynamically without changing the view tree to prevent this from happening.

Reply:
Sounds good, wanted to ask you about this because I did notice some slow down. Happy if you can look into it.